stock_virtual_available_adj/models/product.py

```diff
@@ -19,10 +24,27 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id,
         for product in self.with_context(prefetch_fields=False):
             sale_lines = self.env['sale.order.line'].search(
                 [('product_id', '=', product.id),
-                 ('state', '=', 'sent')])
+                 ('state', 'in', ('sent', 'draft'))])
             if sale_lines:
                 sale_qty = sum(sale_lines.mapped('product_uom_qty'))
+                res[product.id]['draft_sale_qty'] = sum(sale_lines.filtered(
+                    lambda p: p.state == 'draft').mapped('product_uom_qty'))
                 res[product.id]['virtual_available'] = float_round(
                     res[product.id]['virtual_available'] - sale_qty,
                     precision_rounding=product.uom_id.rounding)
         return res
+
+    @api.depends('stock_move_ids.product_qty', 'stock_move_ids.state')
+    def _compute_quantities(self):
+        res = self._compute_quantities_dict(self._context.get('lot_id'),
+                                            self._context.get('owner_id'),
+                                            self._context.get('package_id'),
+                                            self._context.get('from_date'),
+                                            self._context.get('to_date'))
+        for product in self:
+            product.qty_available = res[product.id]['qty_available']
+            product.incoming_qty = res[product.id]['incoming_qty']
+            product.outgoing_qty = res[product.id]['outgoing_qty']
+            product.virtual_available = res[product.id]['virtual_available']
+            if 'draft_sale_qty' in res[product.id]:
+                product.draft_sale_qty = res[product.id]['draft_sale_qty']
```

Comment:
@TimLai125 I would revert _compute_quantities() and _compute_quantities_dict() back to the original (i.e. remove these methods from the custom module), and add individual methods in product model to calculate `draft_sale_qty` and `sent_sale_qty`, for example, and call these methods from `product_archive_cron` and `website_sale_cart_product_check`.  I think the current design is a bit complicated and hard to follow.